src/go/k8s/pkg/resources/secret.go

```diff
@@ -194,6 +194,11 @@ preStopHook () {
 	echo "Setting maintenance mode on node ${NODE_ID}"
 	until [ "${status:-}" = "200" ]; do
 		status=$(%s)
+    if [ "${status:-}" = "404" ]; then
+      echo "Got 404 when enabling maintenance mode, giving up because node does not exist anymore."
+      return 1
```

Comment:
Exiting 1 here will cause the Pod to be SIGKILLed. Even if the node has been decommissioned, I don't think we want to invite that behavior. 

> If either a PostStart or PreStop hook fails, it kills the Container. [1]

[1]: https://kubernetes.io/docs/concepts/containers/container-lifecycle-hooks/#hook-handler-execution

Reply:
thanks!
changed to return `0`